packages/manager/apps/key-management-service/src/data/hooks/useDeleteOkmsService.ts

```diff
@@ -0,0 +1,47 @@
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+  deleteService,
+  getOkmsServiceId,
+  getOkmsServiceIdQueryKey,
+} from '../api/okmsService';
+
+export type DeleteOkmsServiceMutationParams = {
+  resourceName: string;
+};
+
+export const deleteOkmsServiceMutationKey = ['delete-okms-service'];
+
+export type UseDeleteOkmsServiceParams = {
+  ovhSubsidiary: string;
+  onSuccess?: () => void;
+  onError?: (result: ApiError) => void;
+};
+
+export const useDeleteOkmsService = ({
+  onSuccess,
+  onError,
+  ovhSubsidiary,
+}: UseDeleteOkmsServiceParams) => {
+  const queryClient = useQueryClient();
+  const { mutate: terminateService, ...mutation } = useMutation({
```

Comment:
Tanstack not preconise destructuring here

https://tanstack.com/query/latest/docs/eslint/no-rest-destructuring